### PR TITLE
Fix referential sanity check for env vars and relative imports

### PR DESCRIPTION
**Pull request by sjakobi**
Background: https://github.com/dhall-lang/dhall-haskell/issues/1970

### CODE REVIEW

**Comment by sjakobi** [comment]
Does this look about right? If so, I'll prepare a corresponding change to the test suite and the Haskell implementation.

**Comment by Nadrieril** [comment]
Hm, I expected that we'd take files relative to the current directory, and I think that's what dhall-rust does. The OP of the linked issue visibly expected something else. Happy to forbid it, it would probably be less surprising.

**Comment by Gabriel439** [comment]
@Nadrieril: The main reason that I think we should forbid it is because the original spirit of the referential transparency check is that a Dhall expression's result doesn't change regardless of where you import it from.

File: standard/imports.md [hunk 1]
```diff
@@ -447,7 +447,27 @@ both imports have been canonicalized) or `missing` imports:
     referentiallySane(https://authority₀ directory₀ file₀, missing)
 
 
-... whereas non-remote imports can import anything:
+Environment variables mustn't import relative imports, since we don't know the
```
Comment (TristanCacqueray) [inline standard/imports.md:450]: Isn't `must not` more common in formal writing?

**Comment by Gabriel439** [comment]
Closing this as it looks like this has stalled, but feel free to reopen this again if you'd like to revisit this later

**Pull request closed without merging**